api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirConditionServiceImpl.java

```diff
@@ -38,13 +53,133 @@ public class FhirConditionServiceImpl extends BaseFhirService<Condition, org.ope
+	
 	@Getter(value = AccessLevel.PROTECTED)
 	@Setter(value = AccessLevel.PROTECTED, onMethod_ = @Autowired)
 	private SearchQuery<org.openmrs.Condition, Condition, FhirConditionDao, ConditionTranslator<org.openmrs.Condition>, SearchQueryInclude<Condition>> searchQuery;
 	
+	@Getter(value = AccessLevel.PROTECTED)
+	@Setter(value = AccessLevel.PACKAGE, onMethod_ = @Autowired)
+	private FhirDiagnosisService diagnosisService;
+	
+	@Override
+	public Condition get(@Nonnull String uuid) {
+		
+		Condition result;
+		try {
+			result = super.get(uuid);
+		}
+		catch (ResourceNotFoundException e) {
+			result = diagnosisService.get(uuid);
+		}
+		
+		return result;
+	}
+	
+	@Override
+	public Condition create(@Nonnull Condition condition) {
+		
+		if (condition == null) {
+			throw new InvalidRequestException("Condition cannot be null");
+		}
+		
+		FhirUtils.OpenmrsConditionType result = FhirUtils.getOpenmrsConditionType(condition).orElse(null);
+		
+		if (result == null) {
+			throw new InvalidRequestException("Invalid type of request");
+		}
+		
+		if (result.equals(FhirUtils.OpenmrsConditionType.CONDITION)) {
+			return super.create(condition);
+		}
+		
+		if (result.equals(FhirUtils.OpenmrsConditionType.DIAGNOSIS)) {
+			return diagnosisService.create(condition);
+		}
+		
+		throw new InvalidRequestException("Invalid type of request");
+	}
+	
+	@Override
+	public Condition update(@Nonnull String uuid, @Nonnull Condition condition) {
+		
+		if (uuid == null) {
+			throw new InvalidRequestException("Uuid cannot be null.");
+		}
+		
+		if (condition == null) {
+			throw new InvalidRequestException("Condition cannot be null");
+		}
+		
+		FhirUtils.OpenmrsConditionType result = FhirUtils.getOpenmrsConditionType(condition).orElse(null);
+		
+		if (result == null) {
+			throw new InvalidRequestException("Invalid type of request");
+		}
+		
+		if (result.equals(FhirUtils.OpenmrsConditionType.CONDITION)) {
+			return super.update(uuid, condition);
+		}
+		
+		if (result.equals(FhirUtils.OpenmrsConditionType.DIAGNOSIS)) {
+			return diagnosisService.update(uuid, condition);
+		}
+		
+		throw new InvalidRequestException("Invalid type of request");
+	}
+	
+	@Override
+	public void delete(@Nonnull String uuid) {
+		if (uuid == null) {
+			throw new InvalidRequestException("Uuid cannot be null.");
+		}
+		
+		try {
+			super.delete(uuid);
+		}
+		catch (ResourceNotFoundException e) {
+			diagnosisService.delete(uuid);
+		}
+	}
+	
 	@Override
 	public IBundleProvider searchConditions(ConditionSearchParams conditionSearchParams) {
-		return searchQuery.getQueryResults(conditionSearchParams.toSearchParameterMap(), dao, translator,
-		    searchQueryInclude);
+		SearchParameterMap theParams = conditionSearchParams.toSearchParameterMap();
+		
+		IBundleProvider diagnosisBundle = null;
+		IBundleProvider conditionBundle = null;
+		
+		if (shouldSearchExplicitlyFor(conditionSearchParams.getTag(), "diagnosis")) {
```

Comment:
We should be looking at the `category` search parameter not the `_tag` one, since that maps to `Condition.category`.